### ISSUE TITLE
Source for GenerateFrameworkBuildFile.cs

### ISSUE THREAD

**Issue by purkhusid** [Brightspace/rules_csharp#167]
Hey! Would it be possible to get the source for `build\src\D2L.Build.BazelGenerator\Commands\GenerateFrameworkBuildFile.cs` that is mentioned in the generated files in this repo?

**Comment by omsmith** [comment]
Hey @purkhusid!

It's part of a larger project of ours, but will look into extracting it.

**Comment by purkhusid** [comment]
That would be awesome! Thanks!

**Comment by omsmith** [comment]
We've extracted the function out into https://github.com/Brightspace/D2L.Build.BazelGenerator.

We'll have the admit that while the comment screams _Don't edit it by hand!_, the generator is not perfect (though we didn't necessarily try very hard to make it so) and we've had to massage the output. So, surely isn't usable as-is, but might be the basic idea for building something else.

But the basic logic:
* Read the DLLs from the reference assembly package
* Convert assembly references into deps
* Produce an import_library rule for each dll

It doesn't produce aliases which we added in by hand. And I believe there's some differences in the dependencies of what we actually committed to the framework files. Fully generating it was aspirational, but that was also a long time ago, so sorry for being fuzzy on the details.

Hope it helps.
